File: rms_desktop/api_client.py
import os
import requests
from dotenv import load_dotenv

load_dotenv()

BASE_URL = os.getenv('RMS_BACKEND_URL', 'http://127.0.0.1:8000')

class ApiClient:
    def __init__(self, base_url: str = None):
        self.base_url = base_url or BASE_URL
        self.token = None

    def _headers(self):
        headers = {'Content-Type': 'application/json'}
        if self.token:
            headers['Authorization'] = f'Bearer {self.token}'
        return headers

    def login(self, username: str, password: str):
        url = f'{self.base_url}/api/auth/login/'
        response = requests.post(url, json={'username': username, 'password': password})
        response.raise_for_status()
        data = response.json()
        self.token = data.get('access')
        return data

    def refresh_token(self, refresh_token: str):
        url = f'{self.base_url}/api/auth/refresh/'
        response = requests.post(url, json={'refresh': refresh_token})
        response.raise_for_status()
        data = response.json()
        self.token = data.get('access')
        return data

    def get(self, path: str):
        url = f'{self.base_url}{path}'
        response = requests.get(url, headers=self._headers())
        response.raise_for_status()
        return response.json()

    def get_properties(self):
        return self.get('/api/properties/')

    def get_units(self):
        return self.get('/api/units/')

    def get_landlords(self):
        return self.get('/api/landlords/')

    def get_landlord_properties(self):
        return self.get('/api/landlord-properties/')

    def get_tenants(self):
        return self.get('/api/tenants/')

    def get_tenant_units(self):
        return self.get('/api/tenant-units/')

    def get_leases(self):
        return self.get('/api/leases/')

    def get_invoices(self):
        return self.get('/api/invoices/')

    def get_payments(self):
        return self.get('/api/payments/')

    def get_expenses(self):
        return self.get('/api/expenses/')

    def get_maintenance_requests(self):
        return self.get('/api/maintenance-requests/')

File: rms_electron/api-client.js
const axios = require('axios');
const fs = require('fs');
const path = require('path');

// Load environment variables from .env file
function loadEnv() {
  const envPath = path.join(__dirname, '.env');
  if (fs.existsSync(envPath)) {
    const envContent = fs.readFileSync(envPath, 'utf-8');
    const lines = envContent.split('\n');
    lines.forEach(line => {
      const [key, value] = line.split('=');
      if (key && value) {
        process.env[key.trim()] = value.trim();
      }
    });
  }
}

loadEnv();

const BASE_URL = process.env.RMS_BACKEND_URL || 'http://127.0.0.1:8000';

class ApiClient {
  constructor() {
    this.baseURL = BASE_URL;
    this.token = null;
    this.client = axios.create({
      baseURL: this.baseURL,
      headers: {
        'Content-Type': 'application/json'
      }
    });
  }

  getHeaders() {
    const headers = {
      'Content-Type': 'application/json'
    };
    if (this.token) {
      headers['Authorization'] = `Bearer ${this.token}`;
    }
    return headers;
  }

  async login(username, password) {
    const response = await axios.post(`${this.baseURL}/api/auth/login/`, {
      username,
      password
    });
    this.token = response.data.access;
    return response.data;
  }

  async refreshToken(refreshToken) {
    const response = await axios.post(`${this.baseURL}/api/auth/refresh/`, {
      refresh: refreshToken
    });
    this.token = response.data.access;
    return response.data;
  }

  async get(path) {
    const response = await axios.get(`${this.baseURL}${path}`, {
      headers: this.getHeaders()
    });
    return response.data;
  }

  async getProperties() {
    return this.get('/api/properties/');
  }

  async getUnits() {
    return this.get('/api/units/');
  }

  async getLandlords() {
    return this.get('/api/landlords/');
  }

  async getLandlordProperties() {
    return this.get('/api/landlord-properties/');
  }

  async getTenants() {
    return this.get('/api/tenants/');
  }

  async getTenantUnits() {
    return this.get('/api/tenant-units/');
  }

  async getLeases() {
    return this.get('/api/leases/');
  }

  async getInvoices() {
    return this.get('/api/invoices/');
  }

  async getPayments() {
    return this.get('/api/payments/');
  }

  async getExpenses() {
    return this.get('/api/expenses/');
  }

  async getMaintenanceRequests() {
    return this.get('/api/maintenance-requests/');
  }
}

const apiClient = new ApiClient();
